pkg/virt-api/rest/subresource.go

```diff
@@ -375,6 +379,28 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 	name := request.PathParameter("name")
 	namespace := request.PathParameter("namespace")
 
+	var gracePeriodint64 int64
+	var bodyMap map[string]string
```

Comment:
@vatsalparekh I think that this is the reason why the client-python generation does not work.

We should have a proper struct for this one:

```go
// RestartOptions may be provided when deleting an API object.
type RestartOptions struct {
	TypeMeta `json:",inline"`

	// The duration in seconds before the object should be force-restared. Value must be non-negative integer.
	// The value zero indicates, restart immediately. If this value is nil, the default grace period for deletion of the corresponding VMI for the
	// specified type will be used to determine on how much time to give the VMI to restart.
	// Defaults to a per object value if not specified. zero means restart immediately.
        // Allowed Values: nil and 0
	// +optional
	GracePeriodSeconds *int64 `json:"gracePeriodSeconds,omitempty" protobuf:"varint,1,opt,name=gracePeriodSeconds"`
}
```

Reply:
Note that additional fields like `Preconditions` or `DryRun` will make sense on the long run here too: https://github.com/kubernetes/kubernetes/blob/de1cfa9bc16a08116b8f6b56929d3b1ea39fd1e3/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L454